From a70b0591fe9b71c8f974b2b33982965b432845ad Mon Sep 17 00:00:00 2001
From: Alessandro Fontana <alessandro.fontana@pm.me>
Date: Sat, 29 Aug 2026 15:57:53 +0000
Subject: [PATCH] ASoC: samsung: hook up s1402x and gtaxl-rt5659

Add the Kconfig symbols and Makefile rules for the s1402x audio mixer
driver and the gtaxl RT5659 machine driver.
---
 sound/soc/samsung/Kconfig  | 10 ++++++++++
 sound/soc/samsung/Makefile |  4 ++++
 2 files changed, 14 insertions(+)

diff --git a/sound/soc/samsung/Kconfig b/sound/soc/samsung/Kconfig
index ec7204f57f..0620d9eae0 100644
--- a/sound/soc/samsung/Kconfig
+++ b/sound/soc/samsung/Kconfig
@@ -21,6 +21,16 @@ config SND_SAMSUNG_SPDIF
 config SND_SAMSUNG_I2S
 	tristate "Samsung I2S interface support"
 
+config SND_SOC_SAMSUNG_S1402X
+	tristate "Samsung s1402x audio mixer support"
+
+config SND_SOC_SAMSUNG_GTAXL_RT5659
+	tristate "Audio support for gtaxl tablets with RT5659"
+	depends on I2C
+	select SND_SAMSUNG_I2S
+	select SND_SOC_SAMSUNG_S1402X
+	select SND_SOC_RT5659
+
 config SND_SOC_SAMSUNG_SMDK_WM8994
 	tristate "SoC I2S Audio support for WM8994 on SMDK"
 	depends on I2C=y
diff --git a/sound/soc/samsung/Makefile b/sound/soc/samsung/Makefile
index 8d5f091479..67e6bbb932 100644
--- a/sound/soc/samsung/Makefile
+++ b/sound/soc/samsung/Makefile
@@ -5,12 +5,14 @@ snd-soc-idma-y := idma.o
 snd-soc-samsung-spdif-y := spdif.o
 snd-soc-pcm-y := pcm.o
 snd-soc-i2s-y := i2s.o
+snd-soc-s1402x-y := s1402x.o
 
 obj-$(CONFIG_SND_SOC_SAMSUNG) += snd-soc-s3c-dma.o
 obj-$(CONFIG_SND_SAMSUNG_SPDIF) += snd-soc-samsung-spdif.o
 obj-$(CONFIG_SND_SAMSUNG_PCM) += snd-soc-pcm.o
 obj-$(CONFIG_SND_SAMSUNG_I2S) += snd-soc-i2s.o
 obj-$(CONFIG_SND_SAMSUNG_I2S) += snd-soc-idma.o
+obj-$(CONFIG_SND_SOC_SAMSUNG_S1402X) += snd-soc-s1402x.o
 
 # S3C24XX Machine Support
 snd-soc-smdk-wm8994-y := smdk_wm8994.o
@@ -27,6 +29,7 @@ snd-soc-arndale-y := arndale.o
 snd-soc-tm2-wm5110-y := tm2_wm5110.o
 snd-soc-aries-wm8994-y := aries_wm8994.o
 snd-soc-midas-wm1811-y := midas_wm1811.o
+snd-soc-gtaxl-rt5659-y := gtaxl_rt5659.o
 
 obj-$(CONFIG_SND_SOC_SAMSUNG_SMDK_WM8994) += snd-soc-smdk-wm8994.o
 obj-$(CONFIG_SND_SOC_SNOW) += snd-soc-snow.o
@@ -42,3 +45,4 @@ obj-$(CONFIG_SND_SOC_ARNDALE) += snd-soc-arndale.o
 obj-$(CONFIG_SND_SOC_SAMSUNG_TM2_WM5110) += snd-soc-tm2-wm5110.o
 obj-$(CONFIG_SND_SOC_SAMSUNG_ARIES_WM8994) += snd-soc-aries-wm8994.o
 obj-$(CONFIG_SND_SOC_SAMSUNG_MIDAS_WM1811) += snd-soc-midas-wm1811.o
+obj-$(CONFIG_SND_SOC_SAMSUNG_GTAXL_RT5659) += snd-soc-gtaxl-rt5659.o
-- 
2.47.3

